### PR TITLE
fix: :bug:

### PR DESCRIPTION
**Pull request by KanishkaRajputd**
### Summary by CodeRabbit

- New Feature: Introduced a verification check for user identity, improving security and data integrity.
- Refactor: Enhanced the logic of `IdentityForm` and `SocialVerification` components to handle already verified fields, optimizing performance and user experience.
- New Feature: Added a `redirect` prop to `VoteReferendum`, `VoteReferendumEthV2`, `DelegationSuccessPopup`, `DelegateModal`, and `UndelegateModal` components, providing more control over component behavior after successful actions.
- Refactor: Updated `OnChainIdentity` component to handle identity verification status and manage local storage more efficiently, enhancing application stability and performance.

### Summary by CodeRabbit

- Bug Fix: Added a new condition to the verification API endpoint to handle cases where `checkingVerified` is truthy and return the appropriate response.
- Refactor: Modified the `getUserDropDown` function and `AppLayout` component to conditionally truncate the displayed username and changed the initial state of `isIdentityUnverified` to `true`.
- Refactor: Updated the `DelegationSuccessPopup` component to include a `redirect` prop for controlling page reload behavior after closing the popup.
- Refactor: Made changes to the `SocialVerification` component to merge `data` and `field` into a new variable, updated `localStorage.setItem`, and modified the `setSocials` function.
- Refactor: Updated the `OnChainIdentity` component with a new interface, modified state variables and form fields, and made changes to the `getAccounts` function and `handleLocalStorageSetUnverified` function.
- Chore: Added a `redirect` prop to the `VoteReferendum` and `VoteReferendumEthV2` components.
- Chore: Added a `redirect` prop to the `DelegationSuccessPopup` component in the `DelegateModal` and `UndelegateModal` components.

### CODE REVIEW

File: src/components/Post/GovernanceSideBar/Referenda/VoteReferendumEthV2.tsx
Comment (alphainfinitus) [inline src/components/Post/GovernanceSideBar/Referenda/VoteReferendumEthV2.tsx:1]: redirect should be an optional prop with the default value of false

File: src/components/OnchainIdentity/SocialVerification.tsx [hunk 1]
```diff
@@ -200,11 +201,6 @@ const SocialVerification = ({ className, socials, onCancel, setLoading, closeMod
 				}
 			} else if (!checkingVerified || isNotificaiton) {
```
Comment (alphainfinitus) [inline src/components/OnchainIdentity/SocialVerification.tsx:202]: truthy value for isNotificaiton is used but queueNotification is removed

**Review by alphainfinitus — CHANGES_REQUESTED** [review]
Please check comments

**Review by alphainfinitus — CHANGES_REQUESTED** [review]
Please check comments

File: src/components/Listing/Tracks/DelegationSuccessPopup.tsx [hunk 2]
```diff
@@ -59,8 +60,9 @@ const DelegationSuccessPopup = ({
 	ayeVoteValue,
 	nayVoteValue,
 	abstainVoteValue,
-	isVote = false,
-	isMultisig
+	isMultisig,
+	redirect,
```
[the remark this answers is not in the document]
Comment (KanishkaRajputd) [reply src/components/Listing/Tracks/DelegationSuccessPopup.tsx:64]: initially it will be false

**Pull request merged**